### PR TITLE
Improving Crunchbase getters

### LINKED ISSUE #44: Fixing Crunchbase getters

### PR DESCRIPTION
**Pull request by beingkk**
Fixes #44 

Modifies (rewrites) discovery_utils.getters.crunchbase

Adds CrunchbaseGetter class for easy and convenient fetching of the different Crunchbase data tables.

## Instructions to the reviewers

Hey Rosie @RFOxbury, would appreciate if you can take a look at this PR and see that it makes sense! It's largely lifted from innovation_sweet_spots repo, so should be OK, but good to have another pair of eyes.

Hey Shabeer @shabrf and Solomon @sqr00t , no need to review this PR for you - just wanted to ask your opinion on how you would write tests for such a getter class. For example, I guess we could mock up an S3 bucket using LocalStack to check the logic... and then do the actual test of fetching the data from the real S3 bucket and running some data integrity tests? Just curious if you have seen something 

Another question where I'd like your opinion - if I would like to add some utility functions that are not simply getters but say help query the Crunchbase data (eg, "find me all companies belonging to Crunchbase category X"). Do you think it makes sense to still add such a utility function in getters, or should we have a different category of utils for that? (eg, "analysis utils")

Ohh, and I had to downgrade Python version to 3.11 because pflake8 was giving an error - I guess some incompatibility with Python 3.12, not sure why it suddenly appeared!

```
Run Flake8...............................................................Failed
- hook id: flake8
- exit code: 1

Traceback (most recent call last):
  File "/Users/karlis.kanders/Code/discovery_utils/.venv/bin/pflake8", line 5, in <module>
    from pflake8.__main__ import main
  File "/Users/karlis.kanders/Code/discovery_utils/.venv/lib/python3.12/site-packages/pflake8/__init__.py", line 50, in <module>
    class DivertingSafeConfigParser(ConfigParserTomlMixin, configparser.SafeConfigParser):
                                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
```

### CODE REVIEW

File: discovery_utils/getters/crunchbase.py [hunk 1]
```diff
@@ -1,134 +1,289 @@
...
+                return latest_version
+            else:
+                raise ValueError("No valid Crunchbase folders found.")
+        except Exception as e:
+            logger.error(f"Error fetching or processing folder names from S3: {str(e)}")
+            raise
+
+    def _get_table(self, key: str) -> pd.DataFrame:
+        """Download parquet table from S3"""
+        logger.info(f"Downloading debates parquet file: {key}")
```
Comment (beingkk) [inline discovery_utils/getters/crunchbase.py:92]:
```suggestion
        logger.info(f"Downloading parquet file: {key}")
```

File: discovery_utils/getters/crunchbase.py [hunk 2]
```diff
@@ -1,134 +1,289 @@
...
+        except Exception as e:
+            logger.error(f"Error downloading parquet file {key}: {str(e)}")
+            raise
+
+    def _get_cb_table(self, table: str) -> pd.DataFrame:
+        """Get Crunchbase data from S3
+
+        Args:
+            table (str): Table name to download
+            data_version (str, optional): Version of data to download.
```
Comment (beingkk) [inline discovery_utils/getters/crunchbase.py:111]:
```suggestion
            table (str): Table name to download
```

**Review by RFOxbury — APPROVED** [review]
Thanks Karlis! The getters largely make sense. I like the fact that it's set up so that each dataframe gets downloaded once per session and then is locally available.

I tested the getters by loading the class and then using it to access each table like so:
```
from discovery_utils.getters import crunchbase

cb_getter = crunchbase.CrunchbaseGetter()

organisations_df = cb_getter.organisations
organisations_enriched_df = cb_getter.organisations_enriched
...
```

For some of the getters if you expect them to be more widely used beyond Automated Horizon Scanning, it might be nice to add more info in the docstrings? For example:

- It looks like the "degrees" table contains info on which degrees are held by which individuals - are these individuals the CEOs of companies? Are they the same individuals that you can find in the "people" and "people_descriptions" tables?
- Looks like the "jobs" table contains info on which individuals held the most important roles in specific companies, with the fields "started_on" and "ended_on" indicating the dates of their tenure
- The "ipos" table has a few different date columns but it looks like the one that users would most likely be interested in is the "went_public_on" field

**Comment by beingkk** [comment]
Thank you @RFOxbury, I'll add your suggestions in the documentation. I'm slowly writing Wiki entries for the repo, so that's probably a good place (or maybe for Crunchbase we'd do a private google doc, because it's proprietary dataset)

I'll go ahead and merge, thanks!

**Pull request merged**